app/bundles/ProjectBundle/Service/ProjectEntityLoaderService.php

```diff
@@ -0,0 +1,279 @@
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class ProjectEntityLoaderService
+{
+    /** @var array<string, EntityTypeConfig> */
+    private array $entityTypesCache = [];
+
+    public function __construct(private EntityManagerInterface $em, private TranslatorInterface $translator, private ModelFactory $modelFactory, private CorePermissions $security)
+    {
+    }
+
+    /**
+     * @param array<string, EntityTypeConfig> $entityTypes
+     *
+     * @return array<string, array<string, mixed>>
+     */
+    public function getProjectEntities(Project $project, array $entityTypes): array
+    {
+        $results = [];
+
+        foreach ($entityTypes as $entityType => $config) {
+            $repository = $config->model->getRepository();
+
+            $entities   = $repository->createQueryBuilder('e')
+                ->join('e.projects', 'p')
+                ->where('p.id = :projectId')
+                ->setParameter('projectId', $project->getId())
+                ->orderBy('e.dateModified', 'DESC')
+                ->getQuery()
+                ->getResult();
+
+            $results[$entityType] = [
+                'label'    => $config->label,
+                'entities' => $entities,
+                'count'    => count($entities),
+            ];
+        }
+
+        return $results;
+    }
+
+    /**
+     * @return array<string, EntityTypeConfig>
+     */
+    private function getEntityTypes(): array
+    {
+        if (!empty($this->entityTypesCache)) {
+            return $this->entityTypesCache;
+        }
+
+        $allMetadata = $this->em->getMetadataFactory()->getAllMetadata();
+
+        foreach ($allMetadata as $metadata) {
+            $entityClass = $metadata->getName();
+
+            foreach ($metadata->getAssociationMappings() as $association) {
+                if (
+                    ClassMetadataInfo::MANY_TO_MANY === $association['type']
+                    && Project::class === $association['targetEntity']
+                ) {
+                    $shortName  = $metadata->getReflectionClass()->getShortName();
+                    $entityType = $this->normalizeEntityType(strtolower($shortName));
+
+                    $this->entityTypesCache[$entityType] = new EntityTypeConfig(
+                        entityClass: $entityClass,
+                        label: $this->getEntityLabel($entityType),
+                        model: $this->findModelForEntityType($entityType),
+                    );
+
+                    break;
+                }
+            }
+        }
+
+        // Sort entity types alphabetically by label
+        uasort($this->entityTypesCache, fn (EntityTypeConfig $a, EntityTypeConfig $b) => strcasecmp($a->label, $b->label));
+
+        return $this->entityTypesCache;
+    }
+
+    /**
+     * Get entity types filtered by user view permissions.
+     *
+     * @return array<string, EntityTypeConfig>
+     */
+    public function getEntityTypesWithViewPermissions(): array
```

Comment:
Public method should be above the private methods in a class. Please reorder the methods here.